File: LoadDLL.cs
public object LoadDLL(byte[] PluginBytes, string typeName, string methodName, bool hasParamters = false, Dictionary<object, object> paramters = null)
{
    var p = System.Reflection.Assembly.Load(PluginBytes);

    Type[] rootTypes = p.GetTypes();

    var ci = p.CreateInstance(Strings.Split(rootTypes[0].Namespace, ".")(0) + "." + typeName);

    if (hasParamters == true)
    {
        foreach (var paramter in paramters)
            Interaction.CallByName(ci, paramter.Key, CallType.Set, paramter.Value);
    }

    object obj = Interaction.CallByName(ci, methodName, CallType.Method);

    return obj;
}

File: LoadDLL.vb
    Public Function LoadDLL(ByVal PluginBytes As Byte(), ByVal className As String, ByVal methodName As String, Optional ByVal parameters As Dictionary(Of Object, Object) = Nothing) As Object
        Dim p = Reflection.Assembly.Load(PluginBytes)

        Dim rootTypes() As Type = p.GetTypes

        Dim ci = p.CreateInstance(Split(rootTypes(0).Namespace, ".")(0) & "." & className)

        If parameters IsNot Nothing Then
            For Each parameter In parameters
                CallByName(ci, parameter.Key, CallType.Set, parameter.Value)
            Next
        End If

        Dim obj As Object = CallByName(ci, methodName, CallType.Method)

        Return obj
    End Function
